networkit/cpp/auxiliary/test/LoggingTimerGTest.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * SortedListGTest.cpp
+ *
+ *  Created on: 22.07.2025
+ *      Author: Andreas Scharf (andreas.b.scharf@gmail.com)
+ */
+
+#include <thread>
+#include <gtest/gtest.h>
+#include <networkit/auxiliary/Timer.hpp>
+
+class LoggingTimerTest : public ::testing::Test {
+protected:
+    std::streambuf *originalCerrBuf;
+    std::stringstream capturedOutput;
+
+    void SetUp() override {
+        originalCerrBuf = std::cerr.rdbuf();
+        std::cerr.rdbuf(capturedOutput.rdbuf());
+    }
+
+    void TearDown() override { std::cerr.rdbuf(originalCerrBuf); }
+
+    std::string getCapturedOutput() { return capturedOutput.str(); }
+};
+
+TEST_F(LoggingTimerTest, testLogsWhenLabelIsUsed) {
+    Aux::Log::setLogLevel("TRACE");
+    {
+        Aux::LoggingTimer timer("test-timer", Aux::Log::LogLevel::TRACE);
+        std::this_thread::sleep_for(std::chrono::milliseconds(5));
+    }
+    const std::string output = getCapturedOutput();
```

Comment:
You can inline this function into the EXPECT below, no need to create another variable, same in the other tests.